src/Test.sol

```diff
@@ -448,8 +449,471 @@ abstract contract Test is DSTest, Script {
             assertApproxEqRel(a, b, maxPercentDelta);
         }
     }
+
+    /*//////////////////////////////////////////////////////////////
+                              JSON PARSING
+    //////////////////////////////////////////////////////////////*/
+
+   struct RawEIP1559Transaction {
+        string[] arguments;
+        address contractAddress;
+        string contractName;
+        // json value name = function
+        string functionName;
+        bytes32 hash;
+        // json value name = tx
+        RawEIP1559TransactionDetail detail;
+        // json value name = type
+        string contractType;
+    }
+
+    struct RawEIP1559TransactionDetail {
+        AccessList[] accessList;
+        bytes data;
+        address from;
+        bytes gas;
+        bytes nonce;
+        address to;
+        bytes txType;
+        bytes value;
+    }
+
+    struct EIP1559Transaction {
```

Comment:
Do these struct names match anything in the foundry repo? If not, wdyt of calling them `RawEIP1559Tx` and `ParsedEIP1559Tx` (a little more brief + descriptive). Similar change for legacy transaction struct.

Alternatively, `RawType0Tx`, `ParsedType1Tx`, etc. is another idea

Reply:
On second thought, naming them after the actual type number such as `RawType2Tx` might not work because I'm not sure if other networks follow the same conventions. For example Polygon has EIP-1559 support, but is it still considered a "type 2" tx there? Not sure. So I like the current naming convention we now have, marking this resolved, thanks @odyslam